src/main/java/org/reaktivity/nukleus/tls/internal/stream/TlsServerFactory.java

```diff
@@ -412,66 +404,71 @@ private int decodeClientHello(
         int progress,
         int limit)
     {
-        TlsRecordInfoFW tlsRecordInfo = tlsRecordInfoRO.tryWrap(buffer, progress, limit);
+        final TlsStoreInfo newTlsStoreInfo = lookupStore.apply(server.correlationRouteId);
+        final SSLContext sslContext = newTlsStoreInfo == null ? null : newTlsStoreInfo.context;
```

Comment:
Can we do this part in the constructor instead?
Perhaps we could even hoist it to `newStream` and pass `context` to `TlsServer` constructor?